tests/fertiscan/db/update_inspection/test_update_sub_labels.py

```diff
@@ -206,8 +207,84 @@ def test_update_sub_labels(self):
         for expected_updated_item in expected_updated_data:
             self.assertTrue(
                 expected_updated_item in updated_data,
-                f"Expected updated item {expected_updated_item} was not found in the updated data"
+                f"Expected updated item {expected_updated_item} was not found in the updated data",
+            )
+
+    def test_update_sub_labels_with_mismatched_arrays(self):
+        # Create sub labels with mismatched 'fr' and 'en' arrays
+        mismatched_sub_labels = json.dumps(
+            {
+                "instructions": {
+                    "fr": [
+                        "1. Dissoudre 50g dans 10L d'eau.",
+                        "2. Appliquer toutes les 2 semaines.",
+                    ],
+                    "en": [
+                        "1. Dissolve 50g in 10L of water.",
+                        "2. Apply every 2 weeks.",
+                        "3. Store in a cool, dry place.",
+                    ],
+                }
+            }
+        )
+
+        try:
+            # Execute the function with mismatched sub labels
+            self.cursor.execute(
+                "SELECT update_sub_labels(%s, %s);",
+                (self.label_id, mismatched_sub_labels),
             )
+        except Exception as e:
+            self.fail(f"update_sub_labels raised an exception unexpectedly: {e}")
+
+        # Verify that no exception was raised and check the data integrity
+        self.cursor.execute(
```

Comment:
Can you use a 📦 sub_label.some_function instead of using manual queries?

Reply:
Done. The rest will be properly done in https://github.com/ai-cfia/ailab-datastore/issues/158